doc/server-secrets.md

```diff
@@ -0,0 +1,106 @@
+# Server Secrets
+
+It is possible to provide a token or credentials for a number of external
+services. These may be used to lift a rate limit or provide access to
+private resources from a self-hosted instance.
+
+Secrets can be set in  `private/secret.json`. For example:
+
+```
+{
+  "gh_token": "..."
+}
+```
+
+## Bintray
+
+* `bintray_user`
+* `bintray_apikey`
+
+The bintray API [requires authentication](https://bintray.com/docs/api/#_authentication)
+Create an account and obtain a token from the user profile page.
+
+
+## GitHub
+
+* `gh_token`
+
+Because of Github rate limits, you will need to provide a token, or else badges
+will stop working once you hit 60 requests per hour, the
+[unauthenticated rate limit][github rate limit].
+
+You can [create a personal access token][personal access tokens] through the
+Github website. When you create the token, you can choose to give read access
+to your repositories. If you do that, your self-hosted Shields installation
+will have access to your private repositories.
+
+When a `gh_token` is specified, it is used in place of the Shields token
+rotation logic.
+
+[github rate limit]: https://developer.github.com/v3/#rate-limiting
+[personal access tokens]: https://github.com/settings/tokens
+
+* `gh_client_id`
+* `gh_client_secret`
```

Comment:
Am I right that these 2 keys are only relevant to the rotating token pool? If so, I think we can probably omit them from these docs (along with `shieldsIps`) as they aren't relevant to self-hosting users. If they are relevant to a wider audience, what do they do in that context?

Reply:
They could be relevant to _large_ self-hosting installations which require more than 5k badges/min, though that is a fringe case. Probably it's fine to leave a little pointer here to the production docs..